server/libs/common/src/services/server-info.service.ts

```diff
@@ -0,0 +1,134 @@
+import { AssetType } from '@app/database/entities';
+import { Inject, Injectable } from '@nestjs/common';
+import diskusage from 'diskusage';
+import { APP_UPLOAD_LOCATION } from '../constants';
+import { asHumanReadable } from '../utils/human-readable.util';
+
+export interface ServerDiskInfo {
+  diskSize: string;
+  diskUse: string;
+  diskAvailable: string;
+  diskSizeRaw: number;
+  diskUseRaw: number;
+  diskAvailableRaw: number;
+  diskUsagePercentage: number;
+}
+
+export interface ServerUsageInfo {
+  audio: number;
+  other: number;
+  photos: number;
+  videos: number;
+  objects: number;
+  usageRaw: number;
+  usage: string;
+  usageByUser: ServerUserUsageInfo[];
+}
+
+export interface ServerUserUsageInfo {
+  userId: string;
+  videos: number;
+  photos: number;
+  audio: number;
+  other: number;
+  usageRaw: number;
+  usage: string;
+}
+
+export type ServerUsageItems = ServerUsageItem[];
+export interface ServerUsageItem {
+  userId: string;
+  assetType: AssetType;
+  assetCount: number;
+  totalSizeInBytes: number;
+}
+
+export const IAssetRepository = 'AssetRepository';
+
+export interface IAssetRepository {
```

Comment:
Is this still a WIP that `IAssetRepository` is in `server-info.service.ts`?

Reply:
I haven't ported over the asset service yet, which is where these would go, but I can make a placeholder and move them out for now.